src/common/playlist/playlist-manager.js

```diff
@@ -0,0 +1,130 @@
+// @flow
+import KalturaPlayer from '../../kaltura-player';
+import {FakeEvent, Utils} from 'playkit-js';
+import {PlaylistEventType} from './playlist-event-type';
+import getLogger from '../utils/logger';
+import {UIWrapper} from '../ui-wrapper';
+
+class PlaylistManager {
+  static _logger: any = getLogger('PlaylistManager');
+
+  _config = {
+    metadata: {},
+    countdown: {
+      duration: 10,
+      showing: true
+    },
+    options: {
+      autoContinue: true
+    },
+    items: []
+  };
+  _activeItemIndex: number;
+
+  constructor(player: KalturaPlayer, uiWrapper: UIWrapper, options: KPOptionsObject) {
+    this._player = player;
+    this._uiWrapper = uiWrapper;
+    this.addBindings();
+    this._activeItemIndex = 0;
+    this.configure(options.playlist);
+  }
+
+  addBindings() {
+    this._player.addEventListener(this._player.Event.PLAYBACK_ENDED, () => this.playNext(this._config.options.autoContinue));
+  }
+
+  configure(options: KPPlaylistConfigObject) {
+    Utils.Object.mergeDeep(this._config, options);
+    if (this._config.items && this._config.items.find(item => !!item.sources)) {
+      this._player.dispatchEvent(new FakeEvent(PlaylistEventType.PLAYLIST_LOADED, {playlist: this._config}));
+      this._setActiveItem().then(() => {
+        // auto play flow should be handled via the autoContinue config
+        this._player.configure({playback: {autoplay: false}});
+      });
+    }
+  }
+
+  _getTimeToShow() {
+    const activeItem = this._config.items[this._activeItemIndex];
+    return (
+      (activeItem.countdown && activeItem.countdown.timeToShow) ||
+      this._player.duration - ((activeItem.countdown && activeItem.countdown.duration) || this._config.countdown.duration)
+    );
+  }
+
+  _setActiveItem(): Promise<*> {
+    const activeItem = this._config.items[this._activeItemIndex];
+    const prevItem = this._config.items[this._activeItemIndex - 1];
+    const nextItem = this._config.items[this._activeItemIndex + 1];
```

Comment:
@yairans need to add % or check the length

Reply:
@itaykinnrot the pointers will just be undefined in this case and I think that's OK, cause we don't want cyclic next/prev on first/last items in a list.
Think of it like this - on first entry of playlist on the first time you will have a back button control that shows the last item of the playlist.